src/tools/SwitchingFunction.cpp

```diff
@@ -487,7 +487,7 @@ double SwitchingFunction::calculate(double distance,double&dfunc)const {
       dfunc=expression_deriv[t].evaluate();
     } else plumed_merror("Unknown switching function type");
 // this is for the chain rule:
-    dfunc*=invr0;
+    if(type!=nativeq) dfunc*=invr0;
```

Comment:
Instead than this if I would divide for invr0 on line 454
if you can change this I will then merge it, thank!